AIDevGallery/Utils/Ollama.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace AIDevGallery.Utils;
+
+internal record OllamaModel(string Name, string Tag, string Id, string Size, string Modified);
+
+internal class Ollama
+{
+    public static List<OllamaModel>? GetOllamaModels()
+    {
+        try
+        {
+            using (var p = new Process())
+            {
+                p.StartInfo.FileName = "ollama";
+                p.StartInfo.Arguments = "list";
+                p.StartInfo.RedirectStandardOutput = true;
+                p.StartInfo.RedirectStandardError = true;
+                p.StartInfo.UseShellExecute = false;
+                p.StartInfo.CreateNoWindow = true;
+
+                p.Start();
+                p.WaitForExit();
+
+                var lines = new List<string>();
+
+                Task.WhenAny(
+                    Task.Run(() =>
+                    {
+                        while (p.StandardOutput.Peek() > -1)
+                        {
+                            var line = p.StandardOutput.ReadLine();
+                            lines.Add(line ?? string.Empty);
+                        }
+                    }),
+                    Task.Delay(10)).Wait();
+
+                List<OllamaModel> models = new();
+
+                if (lines.Count > 1)
+                {
+                    for (var i = 1; i < lines.Count; ++i)
+                    {
+                        var line = lines[i];
+                        var tokens = line.Split("  ", System.StringSplitOptions.RemoveEmptyEntries | System.StringSplitOptions.TrimEntries);
+                        if (tokens.Length != 4)
+                        {
+                            continue;
+                        }
+
+                        var nameTag = tokens[0].Split(':');
+
+                        models.Add(new OllamaModel(nameTag[0], nameTag[1], tokens[1], tokens[2], tokens[3]));
+                    }
+                }
+
+                return models;
+            }
+        }
+        catch
```

Comment:
Should we handle the presence of ollama or not? This seems too generic. We can easily check the Path:
https://github.com/microsoft/msstore-cli/blob/3801a81b9a955b6af20e1ef695f73c952c57de58/MSStore.CLI/Services/ExternalCommandExecutor.cs#L103-L123
Also, should we have this as a setting if it is not on the Path?

Reply:
I thought about it but I thought to keep it simple and just let the check for ollama fail. Good idea about caching that check, I'll add that